core/src/main/java/jenkins/fingerprints/FingerprintStorage.java

```diff
@@ -30,12 +30,13 @@
 import java.io.IOException;
 
 import hudson.model.Fingerprint;
-import jenkins.util.SystemProperties;
 import org.kohsuke.accmod.restrictions.Beta;
 import org.kohsuke.accmod.Restricted;
 
 /**
  * Pluggable fingerprint storage API for fingerprints.
+ * Note: The ID supplied as arguments in various methods to uniquely identify fingerprints should not contain the
+ * character '|'.
```

Comment:
Why is that? Could the reference implementation simply escape such a character if it is ever used?

Reply:
So would you recommend that I change my comment to not using escaped `|`.
The reason is that I need a safe delimeter for combining instance ID with the fingerprint ID. 
IIUC, if I escape all `|` I could do something like
```
String id= "004|034556"; // Fingerprint's ID
String instanceID = "abcdef"; // safe
String identifier = id.replace("|","\\|") + "|" + instanceID;
// store identifier
// ...
// fetch identifier
String parts = identifier.split("(?<!\\\\)\\|");
String id = parts[0].replace("\\|", "|");
String identifier = parts[1];
```